enginetest/queries/insert_queries.go

```diff
@@ -827,6 +828,24 @@ var SpatialInsertQueries = []WriteQueryTest{
 }
 
 var InsertScripts = []ScriptTest{
+	{
+		// https://github.com/dolthub/dolt/issues/6675
+		Name: "issue 6675: on duplicate rearranged getfield indexes from select source",
+		SetUpScript: []string{
+			"create table xy (x int primary key, y datetime)",
+			"insert into xy values (0,'2023-09-16')",
+		},
+		Assertions: []ScriptTestAssertion{
```

Comment:
Might be worth having a `LOAD DATA` test case in addition to this one, just to be a little closer to the customer's exact invocation.

Reply:
I'll see if I can add a LOAD DATA test. We do execute `dolt import ...` as INSERT queries, though. We bind a generic `INSERT INTO (...) VALUES (0,0,0,...)` with an arbitrary values statement, and then manually swap the source with a `ChannelRowSource` `sql.Node` defined in Dolt that generates rows from a data file, similar to a select statement.

Reply:
So we do have some tests for some LOAD DATA cases, including column subset and column reordering https://github.com/dolthub/go-mysql-server/pull/974